string filename
bool enable_record
---
string result